torch/_subclasses/fake_tensor.py

```diff
@@ -497,3 +479,28 @@ def to_cpu(e):
     # and the cpu inputs should be temporary. just convert outputs to meta
     # and continue
     return tree_map(MetaConverter(), r)
+
+
+# Just for use to allow copying a module to fake tensors,
+# does not apply elsewhere
+class FakeCopyMode(TorchFunctionMode):
+    def __init__(self, fake_mode):
+        self.fake_mode = fake_mode
+
+    def __torch_function__(self, func, types, args=(), kwargs=None):
+        kwargs = kwargs if kwargs else {}
+
+        # clone will get called in Parameter deepcopy
+        if func == torch._C._TensorBase.clone:
+            return func(self.fake_mode.from_tensor(args[0]), **kwargs)
+        elif func == torch.Tensor.__deepcopy__:
+            assert len(args) == 2 and len(kwargs) == 0
+            tensor, memo = args
+
+            if id(tensor) in memo:
+                return memo[id(tensor)]
+
+            return self.fake_mode.from_tensor(tensor)
```

Comment:
Do we want to memoize this?

[See parameter for ref](https://github.com/pytorch/pytorch/blob/master/torch/nn/parameter.py#L57)

Reply:
Is 
```
            if id(tensor) in memo:
                return memo[id(tensor)]
```

isn't that memoizing it ?

Reply:
Based on parameter, we also need to explicitly write the output to memo if we haven't seen it before

Reply:
I think the writing of the `memo` happens here: https://github.com/pytorch/pytorch/blob/6f9b5208dd271579fa0fe76d2a71181937697a4e/torch/nn/parameter.py#L56 right ?

Reply:
In parameter, yes. To me this suggests that if we're hitting deepcopy in fake tensor (meaning that we aren't deep copying Parameter(FakeTensor(...)) but just FakeTensor(...)), then we also need to have that same logic in the fake tensor so that memo gets populated

```suggestion
            result = self.fake_mode.from_tensor(tensor)
            memo[id(self)] = result
            return result
```

Reply:
and `self.fake_mode.from_tensor` already memoizes conversion from Normal Tensor -> Fake.. 

But I guess yea 
```           
 if id(tensor) in memo:
                return memo[id(tensor)]
``` is a no-op here since we're not writing to it.

Reply:
Ah I see. Thanks!